src/Arcus.WebApi.Security.Authentication/SharedAccessKey/SharedAccessKeyAuthenticationAttribute.cs

```diff
@@ -8,29 +8,30 @@ namespace Arcus.WebApi.Security.Authentication.SharedAccessKey
     /// <summary>
     /// Authentication filter to secure HTTP requests with shared access keys.
     /// </summary>
-    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)] 
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
     public class SharedAccessKeyAuthenticationAttribute : TypeFilterAttribute
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="SharedAccessKeyAuthenticationAttribute"/> class.
         /// </summary>
         /// <param name="secretName">The name of the request header which value must match the stored secret with the same name as the header.</param>
         /// <exception cref="ArgumentException">When the <paramref name="secretName"/> is <c>null</c> or blank.</exception>
-        public SharedAccessKeyAuthenticationAttribute(string secretName) : this(secretName, secretName) { }
+        public SharedAccessKeyAuthenticationAttribute(string secretName) : this(secretName, secretName, secretName) { }
```

Comment:
We used to support this here @stijnmoreels but maybe it's better to get rid of this, or did we have concrete demand for this?

We could also provide following overloads instead of this one:
```csharp
public SharedAccessKeyAuthenticationAttribute(string headerName) : this(headerName, null, headerName) { }
public SharedAccessKeyAuthenticationAttribute(string queryParameterName, string secretName) : this(null, queryParameterName, secretName) { }
```

Reply:
Well, it can happen that the name of the secret is not the same as the name of the header/queryParam. So, I guess we still have to support it. It seems unreasonable to force them to be the same.

Reply:
So what is it your proposing here?
Add 2 new overloads as @tomkerkhove suggested or..?

Reply:
@tomkerkhove , any suggestions.

Reply:
> Well, it can happen that the name of the secret is not the same as the name of the header/queryParam. So, I guess we still have to support it. It seems unreasonable to force them to be the same.

It's not really clear to me what your point is here @stijnmoreels? With just one value you're assuming it's the same for query parameter, header & secret or?

Reply:
@stijnmoreels @tomkerkhove I have updated the pull request, please have a look

Reply:
Now that we have 2 approaches to define things I think we should still support it and use the configured name for header-auth only based on a secret with the same name, but mark it as obsolete.

How about this approach?
```suggestion
        [Obsolete("We now support multiple ways to specify the authentication key, we recommend using our overload instead")]
        public SharedAccessKeyAuthenticationAttribute(string secretName) : this(secretName, queryParameterName: null, secretName: secretName) { }
```

Reply:
+1, will change as suggested